unified-prototype/unified-plugin/plugin-kmp/src/main/java/org/gradle/api/experimental/kmp/StandaloneKmpLibraryPlugin.java

```diff
@@ -69,7 +90,7 @@ private void linkDslModelToPluginLazy(Project project, KmpLibrary dslModel) {
                     target.getDependencies()
                 );
                 kotlinTarget.getCompilations().configureEach(compilation -> {
-                    compilation.getCompilerOptions().getOptions().getJvmTarget().set(target.getJvmTarget());
+                    compilation.getCompilerOptions().getOptions().getJvmTarget().set(project.provider(() -> JvmTarget.Companion.fromTarget(target.getJdkVersion().get())));
```

Comment:
Couldn't this be something like:
`target.getJdkVersion().map(value -> JvmTarget.Companion.fromTarget(value))`?